### PR TITLE
Add null check for GDExtension deinitialization

### LINKED ISSUE #87937: GDExtension hot reload does not check for null in `initialization.deinitialize`
### Tested versions

Reproducible in: 4.3.dev [b4e2a24], 4.2.1.stable

### System information

macOS 14.2.1

### Issue description

Godot crashes trying to hot reload GDExtension when `initialization.deinitialize` method is not provided by extension developers.

### Steps to reproduce

1. Configure `.gdextension` file to include `reloadable = true`
2. Build the extension library without providing `initialization.deinitialize`
3. Open Godot
4. Change something in extension, rebuild and replace the library
5. Go back to Godot window

### Minimal reproduction project (MRP)

N/A

### PR DESCRIPTION
**Pull request by tishin**
Fixes #87937

### CODE REVIEW

**Review by AThousandShips — APPROVED** [review]
Looks correct to me, is symmetrical with the `initialize_library` code so seems like a simple oversight

CC @dsnopek

File: core/extension/gdextension.cpp [hunk 1]
```diff
@@ -794,6 +794,9 @@ void GDExtension::deinitialize_library(InitializationLevel p_level) {
 	ERR_FAIL_COND(p_level > int32_t(level_initialized));
 
 	level_initialized = int32_t(p_level) - 1;
+    
+	ERR_FAIL_NULL(initialization.deinitialize);
+    
 	initialization.deinitialize(initialization.userdata, GDExtensionInitializationLevel(p_level));
```
Comment (AThousandShips) [inline core/extension/gdextension.cpp:800]:
```suggestion

	ERR_FAIL_NULL(initialization.deinitialize);

	initialization.deinitialize(initialization.userdata, GDExtensionInitializationLevel(p_level));
```
Some trailing whitespace that needs to be removed

**Comment by AThousandShips** [comment]
Please squash your commits into one, see [here](https://docs.godotengine.org/en/latest/contributing/workflow/pr_workflow.html#the-interactive-rebase)

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 4.2.2.

**Pull request merged**